app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -2060,15 +2055,31 @@ class PluginApi {
    * ```
    */
   addSidebarPanel(func) {
-    addSidebarPanel(func);
+    this._lookupContainer("service:sidebar-state").addPanel(func);
   }
 
   /**
    * EXPERIMENTAL. Do not use.
    * Support for setting a Sidebar panel.
    */
   setSidebarPanel(name) {
-    setSidebarPanel(name);
+    this._lookupContainer("service:sidebar-state").setPanel(name);
+  }
+
+  /**
+   * EXPERIMENTAL. Do not use.
+   * Set combined sidebar section mode. In this mode, sections from all panels are displayed together.
+   */
+  setCombinedSidebarMode() {
+    this._lookupContainer("service:sidebar-state").setCombinedMode();
+  }
+
+  /**
+   * EXPERIMENTAL. Do not use.
+   * Set separated sidebar section mode. In this mode, only sections from the current panel are displayed.
+   */
+  setSeparatedSidebarMode() {
```

Comment:
I'm wondering if we need another API for this on top of `setSidebarPanel`. If someone calls `setSidebarPanel`, I think it would mean they want sidebar in the seperated mode already.

Reply:
Good point, I changed SidebarState that when `setSidebarPanel` is called, then we set separated mode.

Similarly, when combined mode is called, then main panel is set as current

Reply:
Sorry I'm still not following in what scenarios is this API required. When a plugin is looking to switch to the separated mode, wouldn't it just call `setSidebarPanel(name)`?